src/external_connections/next_gen_network.py

```diff
@@ -0,0 +1,92 @@
+#This program reads in and modifies lateral flows from the next generation water model and feeds them
+#into mc_reach.compute_network 
+import os
+import sys
+import time
+import numpy as np
+import argparse
+import pathlib
+import pandas as pd
+from functools import partial
+from itertools import chain, islice
+import next_gen_io
+import nhd_network
+import mc_reach
+
+#The following 2 values are currently hard coded for this test domain
+nts = 720  # number of timestep = 1140 * 60(model timestep) = 86400 = day
+dt_mc = 300.0  # time interval for MC
+
+#Currently tested on the Sugar Creek domain
+ngen_network_df = next_gen_io.read_ngen_network_geojson("./sugar_creek_waterbody_data_subset.geojson")
+
+#Create dictionary mapping each connection ID
+ngen_network_dict = dict(zip(ngen_network_df.ID, ngen_network_df.toID))
+
+waterbody_connections = {}
+
+#Cycle through dictionary and extract only the ID integer values
+for key in ngen_network_dict:
```

Comment:
A more concise (and probably faster) way to replace this for loop might be:
```python
def node_key_func(x):
    return int(x[4:])
waterbody_connections = {node_key_func(k): node_key_func(v) for k, v in ngen_network_dict.items()}
```

Reply:
Thanks. Putting this in.